hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java

```diff
@@ -122,10 +123,19 @@ public ResponseEntity<MgmtArtifact> uploadArtifact(@PathVariable("softwareModule
 
     @Override
     public ResponseEntity<List<MgmtArtifact>> getArtifacts(
-            @PathVariable("softwareModuleId") final Long softwareModuleId) {
+            @PathVariable("softwareModuleId") final Long softwareModuleId,
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_WITH_DOWNLOAD_URL, required = false) final Boolean withDownloadUrl) {
```

Comment:
couldn't we re-use the "representation mode" concept here? It is already used for other collections (e.g. for /rest/v1/actions or /rest/v1/rollouts)

Reply:
Good hint, changed